### PR TITLE
Feature/NIJO-1031 added endpoint for getting invitations sent by me and all invitations…

### PR DESCRIPTION
**Pull request by DFSko**
… to my agency

### CODE REVIEW

File: Api/Controllers/AgentControllers/AgentsController.cs [hunk 1]
```diff
@@ -163,13 +163,39 @@ public async Task<IActionResult> GetInvitationData(string code)
         }
 
 
+        /// <summary>
+        ///    Get agency invitations list
```
Comment (mingazhev) [inline Api/Controllers/AgentControllers/AgentsController.cs:167]: Summary should answer a question "what does method do", like "Gets agency invitations list"

File: Api/Controllers/AgentControllers/AgentsController.cs [hunk 2]
```diff
@@ -163,13 +163,39 @@ public async Task<IActionResult> GetInvitationData(string code)
...
+        /// </summary>
+        [HttpGet("agencies/invitations")]
+        [ProducesResponseType(typeof(List<AgentInvitationInfo>), (int) HttpStatusCode.OK)]
```
Comment (mingazhev) [inline Api/Controllers/AgentControllers/AgentsController.cs:170]: Please add a permission attribute there, I think something like "AgentManagement" will be appropriate

File: Api/Controllers/AgentControllers/AgentsController.cs [hunk 3]
```diff
@@ -163,13 +163,39 @@ public async Task<IActionResult> GetInvitationData(string code)
...
+
+
         /// <summary>
         ///     Gets current agent.
         /// </summary>
         /// <returns>Current agent information.</returns>
         [HttpGet("agents")]
         [ProducesResponseType(typeof(AgentDescription), (int) HttpStatusCode.OK)]
         [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.BadRequest)]
+        [InAgencyPermissions(InAgencyPermissions.ReceiveAgencyInvitations)]
```
Comment (mingazhev) [inline Api/Controllers/AgentControllers/AgentsController.cs:198]: I think this attribute on the wrong place. And if you meant "viewing", lets make it "Observe". Receiving invitations looks like invitations are received by mail or something similar

File: Api/Services/Users/IUserInvitationService.cs [hunk 4]
```diff
@@ -17,5 +19,9 @@ public interface IUserInvitationService
         Task Accept(string invitationCode);
 
         Task<Result<TInvitationData>> GetPendingInvitation<TInvitationData>(string invitationCode, UserInvitationTypes invitationType);
+
+        Task<List<AgentInvitationInfo>> GetInvitationsByAgent(int agentId);
```
Comment (mingazhev) [inline Api/Services/Users/IUserInvitationService.cs:23]: This service is common for admins and agents. Why does it contain agent-specific methods?

**Review by mingazhev — CHANGES_REQUESTED** [review]

File: Api/Controllers/AgentControllers/AgentsController.cs (hunk 1 above)
Comment (DFSko) [reply Api/Controllers/AgentControllers/AgentsController.cs:167]: Fixed

File: Api/Controllers/AgentControllers/AgentsController.cs (hunk 2 above)
Comment (DFSko) [reply Api/Controllers/AgentControllers/AgentsController.cs:170]: Fixed

File: Api/Controllers/AgentControllers/AgentsController.cs (hunk 3 above)
Comment (DFSko) [reply Api/Controllers/AgentControllers/AgentsController.cs:198]: Fixed

File: Api/Services/Users/IUserInvitationService.cs (hunk 4 above)
Comment (DFSko) [reply Api/Services/Users/IUserInvitationService.cs]: Fixed

File: Api/Controllers/AgentControllers/AgentsController.cs [hunk 5]
```diff
@@ -114,9 +114,12 @@ public async Task<IActionResult> RegisterInvitedAgent([FromBody] RegisterInvited
         [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.BadRequest)]
         [MinCounterpartyState(CounterpartyStates.ReadOnly)]
         [InAgencyPermissions(InAgencyPermissions.AgentInvitation)]
-        public async Task<IActionResult> InviteAgent([FromBody] AgentInvitationInfo request)
+        public async Task<IActionResult> InviteAgent([FromBody] SendAgentInvitationRequest request)
         {
-            var (_, isFailure, error) = await _agentInvitationService.Send(request);
+            var agent = await _agentContextService.GetAgent();
+            var agentInvitationInfo = new AgentInvitationInfo(request.RegistrationInfo, request.AgencyId, agent.AgentId, request.Email);
```
Comment (mingazhev) [inline Api/Controllers/AgentControllers/AgentsController.cs:120]: As soon we get all the context for agent in agentContext (including an agency too), agencyId should not be passed from frontend side

Comment (mingazhev) [inline Api/Controllers/AgentControllers/AgentsController.cs:120]: This looks like a domain object inside constructor.
Why not passing the request to service and constructing this object inside the service?

File: Api/Models/Agents/SendAgentInvitationRequest.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+
+        /// <summary>
+        ///     Regular agent personal information.
+        /// </summary>
+        [Required]
+        public AgentEditableInfo RegistrationInfo { get; }
+
+        /// <summary>
```
Comment (mingazhev) [inline Api/Models/Agents/SendAgentInvitationRequest.cs:23]: This should not be there

File: HappyTravel.Edo.Common/Enums/InAgencyPermissions.cs [hunk 7]
```diff
@@ -34,6 +35,7 @@ public enum InAgencyPermissions
             AgencyToChildTransfer |
             ReceiveBookingSummary |
             AgencyBookingsManagement |
```
Comment (mingazhev) [inline HappyTravel.Edo.Common/Enums/InAgencyPermissions.cs:37]: You should add a migration to update rights for agents which have "All" permissions. For reference look at "20201110131233_AddIsActiveToRelation"

**Review by mingazhev — CHANGES_REQUESTED** [review]

File: Api/Controllers/AgentControllers/AgentsController.cs (hunk 5 above)
Comment (DFSko) [reply Api/Controllers/AgentControllers/AgentsController.cs]: Fixed

Comment (DFSko) [reply Api/Controllers/AgentControllers/AgentsController.cs]: Fixed

File: Api/Models/Agents/SendAgentInvitationRequest.cs (hunk 6 above)
Comment (DFSko) [reply Api/Models/Agents/SendAgentInvitationRequest.cs:22]: Fixed

File: HappyTravel.Edo.Common/Enums/InAgencyPermissions.cs (hunk 7 above)
Comment (DFSko) [reply HappyTravel.Edo.Common/Enums/InAgencyPermissions.cs:37]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**